crates/polars-core/src/chunked_array/ops/sort/arg_sort.rs

```diff
@@ -12,24 +12,104 @@ where
         options.multithreaded,
     );
 }
+// Compute the indexes after reversing a sorted array, maintaining
+// the order of equal elements, in linear time. Faster than sort_impl
+//  as we avoid allocating extra memory.
+pub(super) fn reverse_stable_no_nulls<I, J, T>(iters: I, len: usize) -> Vec<IdxSize>
+where
+    I: IntoIterator<Item = J>,
+    J: IntoIterator<Item = T>,
+    T: TotalOrd + Send + Sync,
+{
+    let mut current_start: IdxSize = 0;
+    let mut current_end: IdxSize = 1;
+    let mut flattened_iter = iters.into_iter().flatten();
+    let first_element = flattened_iter.next();
+    let mut rev_idx: Vec<IdxSize> = Vec::with_capacity(len);
+    let mut i: IdxSize;
+    // We traverse the array , comparing consecutive elements.
+    // We maintain the start and end indice of elements with same value.
+    // When we see a new element we push the previous indices in reverse order.
+    // We do a final reverse to get stable reverse index.
+    // Example -
+    // 1 2 2 3 3 3 4
+    // 0 1 2 3 4 5 6
+    // We get start and end position of equal values -
+    // 0 1-2 3-5 6
+    // We insert the indexes of equal elements in reverse
+    // 0 2 1 5 4 3 6
+    // Then do a final reverse
+    // 6 3 4 5 1 2 0
+    match first_element {
+        Some(value) => {
+            let mut previous_element = value;
+            for current_element in flattened_iter {
+                if current_element.tot_cmp(&previous_element) == Ordering::Equal {
+                    current_end += 1;
+                } else {
+                    // Insert in reverse order
+                    i = current_end;
+                    while i > current_start {
+                        i -= 1;
+                        //SAFETY - we allocated enough
+                        unsafe { rev_idx.push_unchecked(i) };
+                    }
+                    current_start = current_end;
+                    current_end += 1;
+                }
+                previous_element = current_element;
+            }
+            i = current_end;
+            while i > current_start {
+                i -= 1;
+                unsafe { rev_idx.push_unchecked(i) };
+            }
+            // Final reverse
+            rev_idx.reverse();
+            rev_idx
+        },
+        None => rev_idx,
```

Comment:
What does this return? An empty Vec?

Reply:
Yes, for arrays of length zero we return an empty vec (From `Vec::with_capacity(0)`).